packages/aws-cdk-lib/core/lib/cfn-resource.ts

```diff
@@ -34,10 +34,10 @@ export interface CfnResourceProps {
  */
 export class CfnResource extends CfnRefElement {
   /**
-   * Check whether the given construct is a CfnResource
+   * Check whether the given object is a CfnResource
    */
-  public static isCfnResource(construct: IConstruct): construct is CfnResource {
-    return (construct as any).cfnResourceType !== undefined;
+  public static isCfnResource(x: any): x is CfnResource {
+    return x !== null && typeof(x) === 'object' && x.cfnResourceType !== undefined;
```

Comment:
Other similar isXxx methods check by `Symbol`, is there any reason not to do the following? I thought it would work as it is more safe and also keeps the consistent signatures.

Because the `cfnResourceType` can be considered unique property?

```ts
const CFN_RESOURCE_SYMBOL = Symbol.for('@aws-cdk/core.CfnResource');
// ...
  public static isCfnResource(x: any): x is CfnResource {
    return x !== null && typeof(x) === 'object' && CFN_RESOURCE_SYMBOL in x;
  }
  // ...
  constructor(scope: Construct, id: string, props: CfnResourceProps) {
    // ...
    Object.defineProperty(this, CFN_RESOURCE_SYMBOL, { value: true });
```

Reply:
Thanks. I'd like to keep this as it is to prevent from possible breaking change unless there is a strong reason to do so.

Reply:
I see. Thanks!